Automattic/jetpack
# Editor: Allow composing block editor posts on WordPress.com.

This PR allows WordPress.com users to access the block editor for Jetpack sites on Calypso via an iframe.

See phase 2 notes at p3fqKv-6Hh-p2.

**Testing Instructions**
* Sandbox the API and apply D27129-code to it.
* Load Calypso either locally or by calypso.live (this feature is only available behind the `jetpack/gutenframe` flag on testing environments, being `wpcalypso`, `horizon`, and `development`).
* Directly load the URL `/block-editor/post/:yourJetpackDomain`.
* Verify the block editor iframe does _not_ load (you will see a gray screen with a `sameorigin` error in the console).
* Load this branch.
* Disconnect and reconnect your site to auto-activate the new module.
* If that doesn't work, activate it from the oldskool modules page at `wp-admin/admin.php?page=jetpack_modules` (click "WordPress.com Block Editor Iframe" towards the bottom).
* Directly load the URL `/block-editor/post/:yourJetpackDomain` again.
* Verify the iframed block editor loads as expected.


#### Proposed changelog entry for your changes:

* Block Editor: WordPress.com users can now compose Block Editor posts for their Jetpack sites without leaving WordPress.com.
